src/main/scala/org/broadinstitute/dsde/firecloud/service/WorkspacePublishingSupport.scala

```diff
@@ -37,4 +40,17 @@ trait WorkspacePublishingSupport extends LibraryServiceSupport {
     workspace.attributes.get(publishedFlag).fold(false)(_.asInstanceOf[AttributeBoolean].value)
   }
 
+  def setWorkspacePublishedStatus(ws: Workspace, publishArg: Boolean, rawlsDAO: RawlsDAO, ontologyDAO: OntologyDAO, searchDAO: SearchDAO)(implicit userToken: WithAccessToken): Future[Workspace] = {
+    rawlsDAO.updateLibraryAttributes(ws.namespace, ws.name, updatePublishAttribute(publishArg)) map { workspace =>
+      val docPublishResult = if (publishArg)
+        Try(publishDocument(workspace, ontologyDAO, searchDAO)).isSuccess
+      else
+        Try(removeDocument(workspace, searchDAO)).isSuccess
+      if (docPublishResult)
+        workspace
+      else
+        throw new FireCloudException(s"Unable to update this workspace, ${ws.namespace}:${ws.name}, to $publishArg in elastic search.")
+    }
```

Comment:
scope creep warning!

If the `updateLibraryAttributes ` succeeds, but the `Try(publishDocument)` or `Try(removeDocument)` fails, we're out of sync. Now that you have these in a `Try`, it would be easy enough to look for failures and reset the published attribute. This is GAWB-1683, so feel free to push it off.

Reply:
Yes, that is a problem. I'm going to add some error logging around this so we can get alerted to it, but leave the real fix for 1683.